File: src/test/java/org/projectodd/nodej/QueryStringTest.java
package org.projectodd.nodej;

import static org.fest.assertions.Assertions.assertThat;

import org.junit.Before;
import org.junit.Test;

public class QueryStringTest extends NodejTestSupport {
    
    @Before
    public void setUp() {
        super.setUp();
        eval("var qs = require('querystring')");
    }

    @Test
    public void testEscape() {
        assertThat(eval("qs.escape('This is a simple & short test.')")).isEqualTo("This+is+a+simple+%26+short+test.");
    }
    
    @Test
    public void testUnescape() {
        assertThat(eval("qs.unescape('This+is+a+simple+%26+short+test.')")).isEqualTo("This is a simple & short test.");
    }
    
    @Test
    public void testStringify() {
        assertThat(eval("qs.stringify({ foo: 'ba r', baz: ['qux', 'quux'], corge: '' })")).isEqualTo("foo=ba+r&baz=qux&baz=quux&corge=");
    }
    
    @Test
    public void testStringifyCustomOptions() {
        assertThat(eval("qs.stringify({foo: 'bar', baz: 'qux'}, ';', ':')")).isEqualTo("foo:bar;baz:qux");
    }
    
    @Test
    public void testParse() {
        eval("var obj = qs.parse('foo=bar&baz=qux&baz=quux&corge')");
        assertThat(eval("obj.foo")).isEqualTo("bar");
        assertThat(eval("obj.baz[0]")).isEqualTo("qux");
        assertThat(eval("obj.baz[1]")).isEqualTo("quux");
        assertThat(eval("obj.corge")).isEqualTo("");
    }
    
    @Test
    public void testParseWithOptions() {
        eval("var obj = qs.parse('foo:bar;baz:qux;baz:quux;corge', ';', ':')");
        assertThat(eval("obj.foo")).isEqualTo("bar");
        assertThat(eval("obj.baz[0]")).isEqualTo("qux");
        assertThat(eval("obj.baz[1]")).isEqualTo("quux");
        assertThat(eval("obj.corge")).isEqualTo("");
    }
}

File: src/test/resources/query_string/query_string_test.js
var vertxTest = require('vertx_tests');
var vassert   = vertxTest.vassert;

var qs = require('querystring');

var QueryStringTests = {
  testEscape: function() {
    vassert.assertEquals("This+is+a+simple+%26+short+test.", qs.escape('This is a simple & short test.'));
    vassert.testComplete();
  },

  testUnescape: function() {
    vassert.assertEquals('This is a simple & short test.', qs.unescape("This+is+a+simple+%26+short+test."));
    vassert.testComplete();
  },

  testStringify: function() {
    vassert.assertEquals("foo=ba+r&baz=qux&baz=quux&corge=", qs.stringify({ foo: 'ba r', baz: ['qux', 'quux'], corge: '' }));
    vassert.testComplete();
  },

  testStringifyCustomOptions: function() {
    vassert.assertEquals("foo:bar;baz:qux", qs.stringify({foo: 'bar', baz: 'qux'}, ';', ':'));
    vassert.testComplete();
  },

  testParse: function() {
    var obj = qs.parse('foo=bar&baz=qux&baz=quux&corge');
    vassert.assertEquals("bar", obj.foo);
    vassert.assertEquals("qux", obj.baz[0]);
    vassert.assertEquals("quux", obj.baz[1]);
    vassert.assertEquals("", obj.corge);
    vassert.testComplete();
  },

  testParseWithOptions: function() {
    var obj = qs.parse('foo:bar;baz:qux;baz:quux;corge', ';', ':');
    vassert.assertEquals("bar", obj.foo);
    vassert.assertEquals("qux", obj.baz[0]);
    vassert.assertEquals("quux", obj.baz[1]);
    vassert.assertEquals("", obj.corge);
    vassert.testComplete();
  }
}

vertxTest.startTests(QueryStringTests);
